buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -57,22 +58,33 @@ class ClusterFormationTasks {
     }
 
     static void addNodeStartupTasks(Project project, Task task, ClusterConfiguration config, File baseDir) {
+        File pidFile = pidFile(baseDir)
         String clusterName = "${task.path.replace(':', '_').substring(1)}"
-        File home = new File(baseDir, "elasticsearch-${ElasticsearchProperties.version}")
-        List setupDependsOn = [project.configurations.elasticsearchZip]
-        setupDependsOn.addAll(task.dependsOn)
-        Task setup = project.tasks.create(name: task.name + '#setup', type: Copy, dependsOn: setupDependsOn) {
-            from { project.zipTree(project.configurations.elasticsearchZip.singleFile) }
-            into baseDir
+        File home = homeDir(baseDir, config.distribution)
+        List setupDeps = [] // need to copy the deps, since start will later be added, which would create a circular task dep!
+        setupDeps.addAll(task.dependsOn)
+        Task setup = project.tasks.create(name: "${task.name}#clean", type: Delete, dependsOn: setupDeps) {
+            delete baseDir
         }
+        setup = configureExtractTask(project, "${task.name}#extract", config.distribution, baseDir, setup)
         // chain setup tasks to maintain their order
-        setup = project.tasks.create(name: "${task.name}#clean", type: Delete, dependsOn: setup) {
-            delete new File(home, 'plugins'), new File(home, 'data'), new File(home, 'logs')
-        }
         setup = project.tasks.create(name: "${task.name}#configure", type: DefaultTask, dependsOn: setup) << {
-            File configFile = new File(home, 'config' + File.separator + 'elasticsearch.yml')
+            File configFile = new File(home, 'config/elasticsearch.yml')
             logger.info("Configuring ${configFile}")
-            configFile.setText("cluster.name: ${clusterName}", 'UTF-8')
+            Map esConfig = [
+                'cluster.name': clusterName,
+                'http.port': config.httpPort,
+                'transport.tcp.port': config.transportPort,
+                'pidfile': pidFile,
+                // TODO: make this work for multi node!
+                'discovery.zen.ping.unicast.hosts': "localhost:${config.transportPort}",
+                'path.repo': "${home}/repo",
+                'path.shared_data': "${home}/../",
+                // Define a node attribute so we can test that it exists
+                'node.testattr': 'test',
+                'repositories.url.allowed_urls': 'http://snapshot.test*'
+            ]
+            configFile.setText(esConfig.collect { key, value -> "${key}: ${value}" }.join('\n'), 'UTF-8')
```

Comment:
do we have an equivalent for the environment properties being set? These are fairly important, at least JAVA_HOME:

https://github.com/elastic/elasticsearch/blob/2.x/dev-tools/src/main/resources/ant/integration-tests.xml#L162-L165